### PR TITLE
Workload sleep/wake from the cli

### PR DESCRIPTION
**Pull request by zerbitx**
**What issue type does this pull request address?** (keep at least one, remove the others)  
/kind bugfix  
/kind enhancement  
/kind test

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible)  
resolves ENGPROV-376

This PR adds end-to-end CLI support for workload sleep/wake flows and fixes the related platform status handling.

It updates `pause` and `resume` so that when a vCluster is configured for workload sleep, the CLI now operates on the workload-sleep state instead of incorrectly falling back to full control-plane sleep/wake behavior. This covers both Helm and Platform drivers, including standalone vClusters.

It also updates platform `list` to show `Sleeping (workloads only)` when the workload-sleep state is present, without sending requests that would wake the cluster during detection.

The implementation includes a shared helper layer for:
- loading and interpreting workload-sleep state
- handling host `vc-config-*` secrets and standalone sleep-state secrets consistently
- using rendered Helm values and resolved project names for platform standalone vClusters
- sending the sleep-ignore header on standalone platform reads

Additional fixes included in this PR:
- preserve and clear the correct sleep annotations during pause/resume
- avoid spurious errors when a non-standalone instance has no cluster ref
- surface malformed config/YAML errors instead of silently falling back
- align standalone detection with `VirtualClusterInstance.Spec.Standalone`

Regression tests were added for Helm and Platform workload sleep/wake flows, list status detection, standalone behavior, and the shared helper edge cases.

**Please provide a short message that should be published in the vcluster release notes**  
Fixed CLI workload sleep/wake handling so `pause`, `resume`, and platform `list` correctly support workload-sleeping vClusters, including standalone platform vClusters.

**What else do we need to know?**  
This change is intentionally scoped to correctness and CLI behavior. It includes shared helper cleanup and test coverage, but does not add list-call parallelization.

### Default Test Execution
The mandatory PR suite runs automatically. Only specify additional test suites below if needed.

### Additional test suites

```label-filter
none

### CODE REVIEW

File: pkg/cli/pause_helm.go [hunk 1]
```diff
@@ -68,6 +84,52 @@ func PauseVCluster(
...
+	sleepingSince := strconv.FormatInt(time.Now().Unix(), 10)
+
+	if vClusterConfig.ControlPlane.Standalone.Enabled {
+		// Annotate the host config secret so find.go can detect StatusWorkloadSleeping,
+		// then also write to the virtual cluster's standalone sleep secret for the
+		// in-cluster sleep state controller.
+		if err := patchSecretWithSleepAnnotations(ctx, kubeClient, vCluster.Namespace, configSecret, sleepingSince, nil); err != nil {
+			return false, err
+		}
+		return true, pauseStandaloneWorkloadSleep(ctx, vCluster, sleepingSince)
```
[the remark this answers is not in the document]
Comment (zerbitx) [reply pkg/cli/pause_helm.go]: I'll add a rollback attempt.

**Comment by zerbitx** [comment]
@codex review

**Comment by zerbitx** [comment]
@codex review

File: pkg/cli/resume_platform.go [hunk 2]
```diff
@@ -54,41 +61,92 @@ func ResumePlatform(ctx context.Context, options *ResumeOptions, config *config.
...
+	return wakePlatformStandaloneTarget(ctx, log, vClusterName, target)
+}
+
+func wakePlatformStandaloneTarget(ctx context.Context, log log.Logger, vClusterName string, target *platformWorkloadSleepSecretTarget) (bool, error) {
+	if target == nil || target.secret == nil {
+		return false, nil
+	}
+
+	log.Infof("Waking standalone vCluster %s workloads (clearing workload sleep mode)", vClusterName)
+	return true, clearSecretSleepAnnotations(ctx, target.kubeClient, target.namespace, target.secret)
```
[the remark this answers is not in the document]
Comment (zerbitx) [reply pkg/cli/resume_platform.go:151]: This ok, it will simply try to clear the sleep annotations. The is sleeping check isn't necessary since it will amount to an empty patch.

File: pkg/cli/resume_helm.go [hunk 3]
```diff
@@ -44,6 +61,87 @@ func ResumeHelm(ctx context.Context, globalFlags *flags.GlobalFlags, vClusterNam
...
+	if err := wakeStandalone(ctx, virtualKubeClient); err != nil {
+		return errWorkloadWake
+	}
+
+	originalConfigSecret := configSecret.DeepCopy()
+	if err := clearSecretSleepAnnotations(ctx, hostKubeClient, namespace, configSecret.DeepCopy()); err != nil {
+		if rollbackErr := wait.PollUntilContextCancel(ctx, time.Second, true, func(ctx context.Context) (bool, error) {
+			if err := restoreStandaloneWorkloadSleep(ctx, virtualKubeClient, originalConfigSecret); err != nil {
+				return false, nil
+			}
```
[the remark this answers is not in the document]
Comment (zerbitx) [reply pkg/cli/resume_helm.go:99]: I'll add a timeout.

File: pkg/cli/pause_helm.go [hunk 4]
```diff
@@ -68,6 +88,73 @@ func PauseVCluster(
...
+	if err := sleepStandaloneWorkload(ctx, virtualKubeClient, sleepingSince); err != nil {
+		return errWorkloadSleep
+	}
+
+	// If the host config patch fails, try to rollback
+	if err := patchSecretWithSleepAnnotations(ctx, hostKubeClient, namespace, configSecret, sleepingSince, nil); err != nil {
+		if rollbackErr := wait.PollUntilContextCancel(ctx, time.Second, true, func(ctx context.Context) (bool, error) {
+			if err := wakeStandalone(ctx, virtualKubeClient); err != nil {
+				return false, nil
+			}
```
[the remark this answers is not in the document]
Comment (zerbitx) [reply pkg/cli/pause_helm.go:136]: I'll add a timeout
